PhysicsTools/HepMCCandAlgos/interface/MCTruthHelper.h

```diff
@@ -563,13 +581,23 @@ namespace MCTruthHelper {
   }
   
   /////////////////////////////////////////////////////////////////////////////
-  const reco::GenParticle *mother(const reco::GenParticle &p) {
-    return static_cast<const reco::GenParticle*>(p.mother());
+  unsigned int numberOfMothers(const reco::GenParticle &p) {
+    return p.numberOfMothers();
+  }
+  
+  /////////////////////////////////////////////////////////////////////////////
+  unsigned int numberOfMothers(const HepMC::GenParticle &p) {
+    return p.production_vertex() ? p.production_vertex()->particles_in_size() : 0;
+  }
+  
+  /////////////////////////////////////////////////////////////////////////////
+  const reco::GenParticle *mother(const reco::GenParticle &p, unsigned int imoth) {
+    return static_cast<const reco::GenParticle*>(p.mother(imoth));
   }
   
   /////////////////////////////////////////////////////////////////////////////
-  const HepMC::GenParticle *mother(const HepMC::GenParticle &p) {
-    return p.production_vertex() && p.production_vertex()->particles_in_size() ? *p.production_vertex()->particles_in_const_begin() : 0;
+  const HepMC::GenParticle *mother(const HepMC::GenParticle &p, unsigned int imoth) {
+    return p.production_vertex() && p.production_vertex()->particles_in_size() ? *(p.production_vertex()->particles_in_const_begin() + imoth) : 0;
   }
```

Comment:
No, imoth is the index of the mother in the list of mothers.  (Yes, genparticles can have more than one mother)